chore(caches): add CI job for testing Redis integration

hypeJunction opened the pull request.

hypeJunction committed:
chore(caches): add CI job for testing Redis integration

jeabakker commented:
the redis travis job keeps failing, i tried restarting a few times. Could you look into this please

hypeJunction committed:
chore(caches): autoload manager cache no longer uses magic methods

Relying on magic __destruct() to store cache was resulting in unexpected
behavior with persistent caches. Autoload manager cache is now stored on
system shutdown event.
Also cleans up some redundant cache flushes.

hypeJunction committed:
Merge branch 'autoload' into redis-job

hypeJunction commented:
Looks like it is.
